### PR TITLE
[Azure] [compute_vm_scaleset] List exposed metrics; add dimension and metric_type metadata

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

- remove duplicated fields
- instead of using wildcard in the field name - use predefined field names
  - for azure.dimensions - dimension name is predefined (`vmname`) - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-compute-virtualmachinescalesets-metrics
  - list of metrics - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-compute-virtualmachinescalesets-metrics
- used fields for dimensions:
  - `cloud.region`
  - `azure.resource.id` - this field includes `azure.subscription_id`, `azure.resource.group`, `azure.resource.name`, format:
     ```
     /subscriptions/{azure.subscription_id}/resourceGroups/{azure.resource.group}/providers/Microsoft.Compute/virtualMachineScaleSets/{azure.resource.name}
     ```

  -  `azure.dimensions.vmname`
  -  `azure.dimensions.virtual_machine`
  -  `azure.timegrain`
- metrics - use predefined metrics - since in the configuration is defined list of metrics, descriptions are taken from https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-compute-virtualmachinescalesets-metrics and https://techcommunity.microsoft.com/t5/ask-the-performance-team/an-overview-of-troubleshooting-memory-issues/ba-p/372673

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: packages/azure_metrics/data_stream/compute_vm_scaleset/fields/fields.yml [hunk 1]
```diff
@@ -1,7 +1,128 @@
...
+      description: Read IOPS from a single disk during monitoring period for OS disk
+    - name: os_disk_write_bytes_per_sec.avg
+      type: float
+      metric_type: gauge
+      description: Bytes/Sec written to a single disk during monitoring period for OS disk
+    - name: os_disk_read_bytes_per_sec.avg
+      type: float
+      metric_type: gauge
+      description: Bytes/Sec read from a single disk during monitoring period for OS disk
+    - name: memory_commit_limit.avg
```
Comment (tetianakravchenko) [inline packages/azure_metrics/data_stream/compute_vm_scaleset/fields/fields.yml:109]: FYI: I couldn't get this metric
available in UI list:
Screenshot 2023-07-28 at 12 21 52

**Review by zmoog — APPROVED** [review]
The changes look good. 

Did you get a chance to collect the missing metric in your tests? I'm not an expert here, but I'm available if you wan to try to get it before merging.

**Comment by tetianakravchenko** [comment]
@zmoog 

> Did you get a chance to collect the missing metric in your tests? I'm not an expert here, but I'm available if you wan to try to get it before merging.

no, I wasn't able to get this one metric `Memory\\Commit Limit`. The problem is that this metric is not even available on the Azure > VM > Diagnostic settings. 

Also I wasn't able to find the full list of metrics, that are supposed to be available in the azure documentation, maybe you know about such documentation?

File: packages/azure_metrics/data_stream/compute_vm_scaleset/fields/package-fields.yml [hunk 2]
```diff
@@ -55,17 +57,14 @@
...
-      type: object
-      object_type: float
-      object_type_mapping_type: "*"
-      description: >
-        Metrics returned.
-
+    - name: dimensions
+      type: group
+      fields:
+        - name: "vmname"
```
Comment (agithomas) [inline packages/azure_metrics/data_stream/compute_vm_scaleset/fields/package-fields.yml:63]: Referring to https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-compute-virtualmachinescalesets-metrics, which field holds LUN value?

Comment (tetianakravchenko) [reply packages/azure_metrics/data_stream/compute_vm_scaleset/fields/package-fields.yml:63]: there is no such field exposed, list of dimensions is defined in configuration - https://github.com/elastic/integrations/blob/main/packages/azure_metrics/data_stream/compute_vm_scaleset/agent/stream/stream.yml.hbs#L71-L73

from my understanding - in this case metric should be grouped by `vmname` only, if only this dimension is defined

**Comment by zmoog** [comment]
> Also I wasn't able to find the full list of metrics, that are supposed to be available in the azure documentation, maybe you know about such documentation?

I can't find this metric either.

I suppose you already checked at https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/metrics-index; I remember this page in your browser's tabs.

**Comment by tetianakravchenko** [comment]
> > Also I wasn't able to find the full list of metrics, that are supposed to be available in the azure documentation, maybe you know about such documentation?
> 
> I can't find this metric either.
> 
> I suppose you already checked at https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/metrics-index; I remember this page in your browser's tabs.

yes, I've seen it, but it does not contain the guest VM metrics there

File: packages/azure_metrics/data_stream/compute_vm_scaleset/fields/package-fields.yml (hunk 2 above)
Comment (tetianakravchenko) [reply packages/azure_metrics/data_stream/compute_vm_scaleset/fields/package-fields.yml:63]: I've added `lun` azure dimension in this PR - https://github.com/elastic/integrations/pull/7314

coult you please have another look on this PR?

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**